rdt/transformers/null.py

```diff
@@ -81,22 +82,22 @@ def transform(self, data):
         Returns:
             numpy.ndarray
         """
-        if self.nulls:
+        if self._null_column or (self.nulls and self._fill_value is not None):
+            isna = data.isna()
             if not isinstance(data, pd.Series):
                 data = pd.Series(data)
 
-            isna = data.isna()
-            if self.nulls and self._fill_value is not None:
-                if not self.copy:
-                    data[isna] = self._fill_value
-                else:
-                    data = data.fillna(self._fill_value)
+        if not self._null_column and (self._fill_value == data.array).any():
+            warnings.warn(IRREVERSIBLE_WARNING)
 
-            if self._null_column:
-                return pd.concat([data, isna.astype('int')], axis=1).to_numpy()
+        if self.nulls and self._fill_value is not None:
```

Comment:
this is used a couple of times. Can we pull it into a variable name? This would also make it easier to understand what this check is for